apps/cevent_frontend/src/app/create_event/_components/DateLocation.tsx

```diff
@@ -0,0 +1,267 @@
+import React, { useState, useCallback, useEffect } from 'react';
+import {
+  Box,
+  Typography,
+  TextField,
+  InputAdornment,
+  Modal,
+} from '@mui/material';
+import { CalendarToday, AccessTime, LocationOn } from '@mui/icons-material';
+import Calendar from 'react-calendar'; // No es necesario importar Value
+import { GoogleMap, Marker, useJsApiLoader } from '@react-google-maps/api';
+import '../_styles/DateLocation.css';
+
+type DateLocationData = {
+  date: Date;
+  time: string;
+  location: string;
+  latitude: number | null;
+  longitude: number | null;
+};
+
+const DateLocation = ({
+  onComplete,
+}: {
+  onComplete: (data: DateLocationData, isComplete: boolean) => void;
+}) => {
+  const [fields, setFields] = useState<DateLocationData>({
+    date: new Date(),
+    time: '',
+    location: '',
+    latitude: null,
+    longitude: null,
+  });
+
+  const [mapState, setMapState] = useState({
+    openDateModal: false,
+    openTimeModal: false,
+    mapCenter: { lat: -17.366198, lng: -66.175489 },
+    markerPosition: { lat: -17.366198, lng: -66.175489 },
+  });
+
+  const { isLoaded } = useJsApiLoader({
+    googleMapsApiKey: process.env.NEXT_PUBLIC_GOOGLE_MAPS_API_KEY || '',
+    libraries: ['places'],
+  });
+
+  useEffect(() => {
+    const isDateValid =
+      fields.date instanceof Date && !isNaN(fields.date.getTime());
+    const isTimeValid =
+      typeof fields.time === 'string' &&
+      /^[0-9]{2}:[0-9]{2}$/.test(fields.time);
+    const isLocationValid =
+      typeof fields.location === 'string' && fields.location.trim() !== '';
+
+    const isComplete = isDateValid && isTimeValid && isLocationValid;
+    onComplete(fields, isComplete);
+  }, [fields, onComplete]);
+
+  const handleMarkerDragEnd = useCallback(
+    async (event: google.maps.MapMouseEvent) => {
+      const lat = event.latLng?.lat();
+      const lng = event.latLng?.lng();
+
+      if (lat !== undefined && lng !== undefined) {
+        const newPosition = { lat, lng };
+        setMapState((prev) => ({
+          ...prev,
+          markerPosition: newPosition,
+        }));
+
+        try {
+          const response = await fetch(
```

Comment:
I could suggest to change the fetching logic to a different file and define it as a service, i.e. to have in Services a MapaService.ts that has a function GetPointInfo = (lat:number, long: number): PointInfo => {} 
All this to separate data fetching logic from UI logic.